cli/azd/pkg/input/console.go

```diff
@@ -84,6 +110,129 @@ func (c *AskerConsole) Message(ctx context.Context, message string) {
 	}
 }
 
+func (c *AskerConsole) MessageUx(ctx context.Context, message string, format MessageUxType) {
+	formattedText, err := addFormat(message, format)
+	// Message and MessageUx don't return errors. Let's log the error and use the original message on error
+	if err != nil {
+		log.Printf("Failed adding format for MessageUx: %s. Using message with no ux format", err.Error())
+		c.Message(ctx, message)
+		return
+	}
+
+	// backwards compatibility to error messages
+	// Remove any formatter before printing the Result
+	// This is can be changed in the future if we want to format any error message as Json or Table when user set output.
+	if format == ResultError {
+		c.formatter = &output.NoneFormatter{}
```

Comment:
This seems worrying. We shouldn't be changing the `formatter` here. Currently this might work since it's the last message, but this could result in undefined behavior if someone issued a `ResultError` midway through command execution.

I'd suggest instead to use `fmt.Println(writer, formattedText)` or refactor that if desired

Reply:
@vhvb1989 Would love to see this addressed, otherwise looks good

Reply:
sure